jmclient/jmclient/taker.py

```diff
@@ -328,7 +328,9 @@ def prepare_my_bitcoin_data(self):
             #find sufficient utxos extremely rare. Indeed, a doubling of 'normal'
             #txfee indicates undesirable behaviour on maker side anyway.
             self.total_txfee = estimate_tx_fee(3, 2,
-                txtype=self.wallet_service.get_txtype()) * self.n_counterparties
+                txtype=self.wallet_service.get_txtype()) * (self.n_counterparties - 1) + \
+                estimate_tx_fee(3, 2, txtype=self.wallet_service.get_txtype(),
+                                outtype=self.wallet_service.get_outtype(self.coinjoin_address()))
```

Comment:
I think this is actually the wrong choice (and the same for line 358 below).

The reason is pretty simple: in a coinjoin, we have a large number of change-outs and coinjoin-outs. *If* a user chooses to do a coinjoin where the destination address they choose has a different scriptPubKey type - and it's rare, and disrecommended, to do this, since it flags the destination as not part of the coinjoin - then still *almost* all of the outputs (default 9 counterparties can mean about 20 outputs) have the wallet scriptpubkey type, not this other scriptpubkey type, so we'd be far better sticking with the default for `outtype`.

Reply:
Not sure if you parsed the line correctly. If the coinjoin output of the taker is the same as the transaction type, then it has the same effect as though the modification never took place; however, if the coinjoin output by the taker is different, then this allows for a more accurate estimation of the fees since the different coinjoin output only comes from the taker. To be fair, these are estimates not the actual fees that would have to be deducted from the transaction so I don't lean any particular way.

Reply:
Oh I see, this and some later comments I've made are just wrong; you're adding the `OUTPUT_EXTRA_BYTES` entry only once. So indeed that works, sorry for confusion.

Of course we may have need for more generality later, but for now this seems fine and is correct.